src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReduceIndexWriteOperation.cs

```diff
@@ -43,6 +43,14 @@ public override void Commit(IndexingStatsScope stats)
                     enqueue.GetAwaiter().GetResult();
                 }
             }
+            catch (TaskCanceledException)
+            {
+                throw;
+            }
+            catch (ObjectDisposedException e) when (DocumentDatabase.DatabaseShutdown.IsCancellationRequested)
+            {
+                throw new TaskCanceledException("A task was canceled", e);
```

Comment:
Can have something meaningful here - e.g. "The operation of writing output reduce documents was cancelled because of database shutdown"?